File: programmers/타겟넘버.js
var answer=0;
function solution(numbers, target) {
    dfs(numbers,target,0);
    return answer;
}
function dfs(numbers,target,index){
    if(index === numbers.length){
        var sum=0;
        for(let i of numbers){
            sum+=i;
        }
        if(target === sum){
            answer++;
        }
        return;
    }else{
        numbers[index]*=1;
        dfs(numbers,target,index+1);
        
        numbers[index]*=-1;
        dfs(numbers,target,index+1);
    }
}

File: programmers/타겟넘버.java
class Solution {
    static int answer = 0;
    public int solution(int[] numbers, int target) {
        dfs(numbers,target,0);
        return answer;
    }
    public void dfs(int[] numbers, int target,int index){
        if(index == numbers.length){
            int sum=0;
            for(int i=0;i<numbers.length;i++){
                sum+=numbers[i];
            }
            if(sum==target){
                answer++;
            }
            return;
        }else{
            numbers[index]*=1;
            dfs(numbers,target,index+1);
            
            numbers[index]*=-1;
            dfs(numbers,target,index+1);
        }
    }
}